### PR TITLE
Added doctest & docstring to quadratic_probing.py

### PR DESCRIPTION
**Pull request by Suyashd999**
### Describe your change:

Contributes to #9943 

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [x] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [x] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [x] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

File: data_structures/hashing/quadratic_probing.py [hunk 1]
```diff
@@ -12,6 +12,40 @@ def __init__(self, *args, **kwargs):
...
+        2. Create hash table with size 8
+        >>> qp = QuadraticProbing(8)
+        >>> qp.insert_data(0)
+        >>> qp.insert_data(999)
+        >>> qp.insert_data(111)
+        >>> qp.keys()
+        {0: 0, 7: 999, 3: 111}
+
+        reference:
+            - https://en.wikipedia.org/wiki/Quadratic_probing
```
Comment (cclauss) [inline data_structures/hashing/quadratic_probing.py:46]: Let's put the URL above the tests so visitors can look at the tests and implementation without any visual clutter.

Comment (Suyashd999) [reply data_structures/hashing/quadratic_probing.py]: @cclauss I made the changes as mentioned by you

File: data_structures/hashing/quadratic_probing.py [hunk 2]
```diff
@@ -12,6 +12,39 @@ def __init__(self, *args, **kwargs):
...
+        {11: 45, 14: 99, 7: 24, 0: 340, 5: 73, 6: 90, 8: 7}
+
+        2. Create hash table with size 8
+        >>> qp = QuadraticProbing(8)
+        >>> qp.insert_data(0)
+        >>> qp.insert_data(999)
+        >>> qp.insert_data(111)
+        >>> qp.keys()
+        {0: 0, 7: 999, 3: 111}
+        """
```
Comment (cclauss) [inline data_structures/hashing/quadratic_probing.py:46]:
```suggestion

        3. Try to add three data elements when only two were provisioned
        >>> qp = QuadraticProbing(2)
        >>> qp.insert_data(0)
        >>> qp.insert_data(999)
        >>> qp.insert_data(111)
        ???
        """
```

Comment (Suyashd999) [reply data_structures/hashing/quadratic_probing.py:62]: > 

Done

**Review by cclauss — APPROVED** [review]

**Pull request merged**